extensions/pkg/controller/backupbucket/errorcodedetector.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupbucket
+
+// ErrorCodeDetector determines the Gardener error codes for errors.
+type ErrorCodeDetector interface {
```

Comment:
You can also enhance the PR description and provider details about the motivation for this change (that currently we have calls to `gardencorev1beta1helper.DetermineError` from the extension library and we need a mechanism for extensions to provider their own error code mapping/detection).
You could also link an example change on how the adaption looks like on the extension side.

Reply:
I will add an example with provider-aws soon.

Reply:
It would look like this: https://github.com/gardener/gardener-extension-provider-aws/compare/master...shafeeqes:enh/ReworkErrorCodes-3

I would add it properly to the description once I open the PR.